contracts/strategies/examples/donation-voting/DonationVotingOnchain.sol

```diff
@@ -198,15 +147,19 @@ contract DonationVotingOnchain is BaseStrategy, RecipientsExtension, Native {
 
     /// @notice Distributes funds (tokens) to recipients.
     /// @param _recipientIds The IDs of the recipients
+    /// @param _data The data containing the allocation token
     /// @param _sender The address of the sender
-    function _distribute(address[] memory _recipientIds, bytes memory, address _sender)
+    function _distribute(address[] memory _recipientIds, bytes memory _data, address _sender)
         internal
         virtual
         override
         onlyAfterAllocation
     {
         if (totalPayoutAmount == 0) totalPayoutAmount = poolAmount;
 
+        (address allocationToken) = abi.decode(_data, (address));
+        if (!_isAllowedToken(allocationToken)) revert TOKEN_NOT_ALLOWED();
```

Comment:
Can't we get the allocation token from the extension? isn't this stored in the strategy, instead of providing it as an argument

Reply:
We store them in a mapping so it's not possible to access without a key.